src/App.js

```diff
@@ -10,15 +10,24 @@ import { RequireAuth } from './components/RequireAuth';
 import Dashboard from './pages/Dashboard';
 import { useDispatch } from 'react-redux';
 import { useEffect } from 'react';
-import { setIsAuth } from './features/userSlice';
+import { handleLogin, handleLogout } from './features/userSlice';
+import jwt_decode from 'jwt-decode';
 
 function App() {
   const dispatch = useDispatch();
 
   useEffect(() => {
     const token = localStorage.getItem('jwt');
 
```

Comment:
Suggestion
```
    const token = localStorage.getItem('jwt');
    if (!token) return;
    const { exp } = jwt_decode(token);
    exp > Date.now() ? dispatch(handleLogin(token)) : dispatch(handleLogout());
```
And `localStorage.removeItem('jwt');` to be handled in userSlice -> `handleLogout()`